pkg/httpapi/api.go

```diff
@@ -261,6 +261,21 @@ func (a *APIRouter) GetApplicationHistory(w http.ResponseWriter, r *http.Request
 		http.Error(w, fmt.Sprintf("failed to get list of application, err: %v", err), http.StatusBadRequest)
 		return
 	}
+	// At this point, if there are no apps as generated by KAM, then check if there are any
+	// apps in general, where the app name is the environment name. Users aren't expected to use both
+	// KAM generated apps and custom apps in their GitOps repo.
+	if len(appList.Items) == 0 {
+		listOptions = nil
+		listOptions = append(listOptions, ctrlclient.InNamespace(""), ctrlclient.MatchingFields{
+			"metadata.name": fmt.Sprintf("%s", envName),
```

Comment:
Why do we need to restrict the app name to match the env name? I wonder if it's possible to use the destination namespace instead `spec.destination.namespace`

Reply:
The UI is passing this to the backend, just like the original case, on line 255.  We want to continue to use that.

Reply:
Yi and I don't see the duplicate entry using 4.12.

Reply:
How did you create the app taxi application?  I'm looking at https://github.com/chetan-rns/argocd-example-apps

Reply:
I created two namespaces dev and stage with the annotation `app.openshift.io/vcs-uri` and created Argo CD Applications to deploy resources in these namespaces

https://gist.github.com/chetan-rns/a1d14fd4c416d7c44176c035241f2b1e

Reply:
We tested again on a 4.11 cluster and there was no duplicate.